crates/config/src/unresolved_config.rs

```diff
@@ -96,12 +98,27 @@ pub struct UnresolvedLogs {
     pub log_level: LogLevel,
 }
 
+impl Default for UnresolvedLogs {
+    fn default() -> Self {
+        UnresolvedLogs {
+            report_hashrate: default_report_hashrate(),
+            log_level: default_log_level()
+        }
+    }
+}
+
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct State {
     #[serde(default = "default_state_path")]
```

Comment:
Perhaps, it is not needed anymore, if whole section is optional.

Reply:
unfortunately, it's needed for a partial case like:
```toml
[state]
```
it will use this default, not the outer, unfortunately. I don't like this repetitiveness, would love a better solution.

and for logs, more cases:
```toml
[logs]
log-level = "trace"
```

Reply:
Yep, logs is different story.